src/Plutarch/Test/QuickCheck/Instances.hs

```diff
@@ -0,0 +1,974 @@
+    Config (..),
+    PlutusType,
+    S,
+    Term,
+    TracingMode (DoTracing),
+    compile,
+    pcon,
+    plam,
+    pmatch,
+    pto,
+    (#),
+    (#$),
+ )
+import Plutarch.Api.V1 (
+    AmountGuarantees (NonZero),
+    PCredential (PPubKeyCredential, PScriptCredential),
+    PCurrencySymbol (PCurrencySymbol),
+    PExtended (PFinite, PNegInf, PPosInf),
+    PInterval (PInterval),
+    PLowerBound (PLowerBound),
+    PMap (PMap),
+    PStakeValidatorHash (PStakeValidatorHash),
+    PTokenName (PTokenName),
+    PUpperBound (PUpperBound),
+    PValidatorHash (PValidatorHash),
+    PValue (PValue),
+ )
+import qualified Plutarch.Api.V1.AssocMap as Assoc (
+    pfilter,
+    pmap,
+ )
+import Plutarch.Api.V1.Time (PPOSIXTime (PPOSIXTime))
+import Plutarch.Api.V1.Tuple (
+    PTuple,
+    pbuiltinPairFromTuple,
+    ptuple,
+    ptupleFromBuiltin,
+ )
+import qualified "plutarch" Plutarch.Api.V1.Value as Value (
+    pnormalize,
+ )
+import Plutarch.Api.V2 (
+    AmountGuarantees (NoGuarantees, Positive),
+    KeyGuarantees (Sorted, Unsorted),
+    PAddress (PAddress),
+    PMaybeData (PDJust, PDNothing),
+    PPubKeyHash (PPubKeyHash),
+    PStakingCredential (PStakingHash, PStakingPtr),
+ )
+import Plutarch.Builtin (ppairDataBuiltin)
+import Plutarch.Evaluate (evalScript)
+import Plutarch.Extra.TermCont (pletC)
+import Plutarch.Lift (PUnsafeLiftDecl (PLifted), plift)
+import qualified Plutarch.List
+import Plutarch.Maybe (pfromJust)
+import Plutarch.Positive (PPositive, ptryPositive)
+import Plutarch.Prelude (
+    PAsData,
+    PBool,
+    PBuiltinList,
+    PBuiltinPair,
+    PByteString,
+    PEither (PLeft, PRight),
+    PEq ((#==)),
+    PInteger,
+    PIsData,
+    PIsListLike,
+    PLift,
+    PList,
+    PListLike (pcons, phead, pnil, pnull, ptail),
+    PMaybe (PJust, PNothing),
+    POrd,
+    PPair (PPair),
+    PPartialOrd ((#<), (#<=)),
+    PRational (PRational),
+    PString,
+    PUnit,
+    Type,
+    pconstant,
+    pdata,
+    pdcons,
+    pdenominator,
+    pdnil,
+    pfield,
+    pfix,
+    pfromData,
+    pfstBuiltin,
+    phoistAcyclic,
+    pif,
+    plet,
+    pnot,
+    pnumerator,
+    pshow,
+    psingleton,
+    psndBuiltin,
+    ptraceError,
+    (:-->),
+ )
+import Plutarch.Show (PShow)
+import Plutarch.TermCont (unTermCont)
+import Test.QuickCheck (
+    Arbitrary (arbitrary, shrink),
+    CoArbitrary (coarbitrary),
+    Function (function),
+    Gen,
+    Testable (property),
+    chooseInt,
+    elements,
+    frequency,
+    functionMap,
+    sized,
+    variant,
+    vectorOf,
+ )
+
+-- | @since 2.0.0
+instance Testable (TestableTerm PBool) where
+    property (TestableTerm t) = property (plift t)
+
+{- | TestableTerm is a wrapper for closed Plutarch term. This
+     abstraction allows Plutarch values can be generated via QuickCheck
+     Generator.
+
+     Hint: Typechecker is picky about how TestableTerm is constructed.
+     Meaning, TestableTerm throw error when it's composed.
+
+ @since 2.0.0
```

Comment:
'closed Plutarch terms'
'allows Plutarch values to be generated via QuickCheck generators'
"The type check is picky about how 'TestableTerm's are constructed."

You probably want `= Note` rather than 'Hint:' for better Haddocks. I'm also not entirely sure what you mean by 'Meaning, TestableTerm throw error when it's composed'.